packages/frontend-web/src/app/scenes/Settings/components/OAuthConfig.tsx

```diff
@@ -0,0 +1,140 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useCallback, useState } from 'react';
+
+import {
+  Button,
+  makeStyles,
+  TextField,
+} from '@material-ui/core';
+
+import { ContainerHeader, OverflowContainer } from '../../../components/OverflowContainer';
+import { Scrim } from '../../../components/Scrim';
+import { API_URL } from '../../../config';
+import { IApiConfiguration } from '../../../platform/dataService';
+import { SCRIM_STYLE } from '../../../styles';
+import { css } from '../../../utilx';
+
+export interface IOAuthConfigProps extends IApiConfiguration {
+  onClickDone(config: IApiConfiguration): any;
+  showLogoutWarning?: boolean;
+}
+
+const useStyles = makeStyles((_theme) => ({
+  textField: {
+    width: '40vw',
+  },
+}));
+
+export function OAuthConfig(props: IOAuthConfigProps) {
+  const [id, setId] = useState(props.id);
+  const [secret, setSecret] = useState(props.secret);
+  const classes = useStyles({});
+
+  const idHandler = useCallback((e) => setId(e.target.value), [setId]);
+  const secretHandler = useCallback((e) => setSecret(e.target.value), [setSecret]);
+  function submitHandler() {
+    props.onClickDone({id, secret});
+  }
+
+  return (
+    <div>
+      <p>Configure your connection to the Google Authentication service.</p>
+      <p>
+        To allocate and view tokens, visit <a
+          href="https://console.developers.google.com/apis/credentials"
+          target="_blank"
+          style={{color: 'inherit', fontWeight: 'bold', textDecoration: 'underline'}}
+        >the Google API console</a>.
+        If you haven't yet got an appropriate <b>OAuth Client ID</b> yet, or want to allocate a new one,
+        you can do so by clicking the <b>Create Credentials</b> button and selecting the <b>OAuth</b> entry.
+      </p>
+      <p>
+        When asked for an application type, choose <b>Web application</b>.
+        Set the <b>Authorised redirect URI</b> to <b>{API_URL}/auth/callback/google</b>.
+        (If you want to connect to YouTube, you'll also need to add a second redirect URI
+        of <b>{API_URL}/youtube/callback</b>.)
+      </p>
+      <p>
+        Once allocated, copy the client id and secret into the fields below.
+      </p>
+      <div style={{textAlign: 'center'}}>
+        <TextField
+          className={classes.textField}
+          label="Client ID"
+          value={id}
+          onChange={idHandler}
+          margin="normal"
+          variant="outlined"
+        />
+        <TextField
+          className={classes.textField}
+          label="Client Secret"
+          value={secret}
+          onChange={secretHandler}
+          margin="normal"
+          variant="outlined"
+        />
+      </div>
+      {props.showLogoutWarning &&
```

Comment:
this && seems weird.

Reply:
This is kinda the idiomatic way of conditionally including bits of a react tree.  See e.g., 

https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

And I agree with you - I really don't like it either.